src/core/io/src/4C_io_input_field.hpp

```diff
@@ -82,6 +88,22 @@ namespace Core::IO
 
     std::variant<T, std::unordered_map<IndexType, T>> data_;
   };
+
+  template <typename T>
+  void read_input_field_from_json(const std::string& file_name, T& value, const std::string& key)
```

Comment:
- If you look at the arguments of this functions, you can see that it is not really related to the `InputField`. I think it would better fit into the `4C_io_yaml.hpp`. You can also see this from the includes, which now pull in a lot of things that are not relevant if you already have an `InputField`.
- Also, the "json" in the name is unnecessary, as this works for yaml and json.
- This function could be improved with a few checks: does the file exist? Is it the correct format? What If the requested key does not exist?